api/app/controllers/spree/api/v1/taxonomies_controller.rb

```diff
@@ -45,13 +45,13 @@ def destroy
         private
 
         def taxonomies
-          @taxonomies = Taxonomy.accessible_by(current_ability).order('name').includes(root: :children).
+          @taxonomies = Taxonomy.accessible_by(current_ability).for_store(current_store).order('name').includes(root: :children).
                         ransack(params[:q]).result.
                         page(params[:page]).per(params[:per_page])
         end
 
         def taxonomy
-          @taxonomy ||= Taxonomy.accessible_by(current_ability, :show).find(params[:id])
+          @taxonomy ||= Taxonomy.accessible_by(current_ability, :show).for_store(current_store).find(params[:id])
```

Comment:
@szymoniwacz let's not modify API v1 controllers, they are going away :)

Reply:
@damianlegawiec Currently Taxons page and product taxons search in admin panel are using API::V1::TaxonsController.
Should I change that to use Admin::TaxonsController with this task?

Reply:
@szymoniwacz that one will be handled in https://github.com/spree/spree/pull/11084